instrumentation/kafka/kafka-clients/kafka-clients-0.11/testing/src/main/java/io/opentelemetry/instrumentation/kafka/internal/AbstractOpenTelemetryMetricsReporterTest.java

```diff
@@ -134,10 +134,18 @@ protected Map<String, Object> consumerConfig() {
     consumerConfig.merge(
         CommonClientConfigs.METRIC_REPORTER_CLASSES_CONFIG,
         TestMetricsReporter.class.getName(),
-        (o, o2) -> o + "," + o2);
+        AbstractOpenTelemetryMetricsReporterTest::mergeValue);
     return consumerConfig;
   }
 
+  @SuppressWarnings("unchecked")
+  private static Object mergeValue(Object o1, Object o2) {
+    List<Object> result = new MetricsReporterList<>();
+    result.addAll((List<Object>) o1);
+    result.add(o2);
+    return result;
+  }
```

Comment:
I didn't follow why test needs to use the internal `MetricsReporterList`?

Reply:
Because we have a test that tries to serialize the configuration https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/17b6cad16fde47f6e2e4aee8e5ddb0cfe1ffc9bb/instrumentation/kafka/kafka-clients/kafka-clients-2.6/library/src/test/java/io/opentelemetry/instrumentation/kafkaclients/v2_6/internal/OpenTelemetryMetricsReporterTest.java#L132 and that test gets the configuration from this method. If we just use an `ArrayList` here then the configuration will contain the list that includes `OpenTelemetryMetricsReporter.class` which will fail in https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/17b6cad16fde47f6e2e4aee8e5ddb0cfe1ffc9bb/instrumentation/kafka/kafka-clients/kafka-clients-2.6/library/src/test/java/io/opentelemetry/instrumentation/kafkaclients/v2_6/internal/OpenTelemetryMetricsReporterTest.java#L159 This is a bit convoluted, but the current behavior that depends on the thread context class loader being able to load the `OpenTelemetryMetricsReporterTest` isn't ideal either.